ci/build-check.sh

```diff
@@ -15,6 +15,9 @@ if test -x /usr/bin/gnome-desktop-testing-runner; then
 fi
 
 if test -x /usr/bin/clang; then
+    # always fail on warnings; https://github.com/ostreedev/ostree/pull/971
+    # Except for clang-4.0: error: argument unused during compilation: '-specs=/usr/lib/rpm/redhat/redhat-hardened-cc1' [-Werror,-Wunused-command-line-argument]
+    export CFLAGS="Wno-error=unused-command-line-argument -Werror ${CFLAGS:-}"
```

Comment:
I think this may be causing `configure` to barf. Let's artifact `config.log` as well for a closer look?